src/script/lua_api/l_object.cpp

```diff
@@ -1888,25 +1888,75 @@ int ObjectRef::l_get_sky(lua_State *L)
 
 	SkyboxParams skybox_params = player->getSkyParams();
 
-	push_ARGB8(L, skybox_params.bgcolor);
-	lua_pushlstring(L, skybox_params.type.c_str(), skybox_params.type.size());
+	if (lua_isboolean(L, 2) && readParam<bool>(L, 2) == true) {
```

Comment:
`readParam<bool>(L, 2, false)` is a shorter way to do this
It would also be better to structure the function like this:
```cpp
  if (!readParam<bool>(L, 2, false)) {
    // deprecated version
    [...]
    return 4;
  }

  [...]
```